http-server/src/main/java/io/micronaut/http/server/exceptions/response/DefaultHtmlErrorResponseBodyProvider.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.exceptions.response;
+
+import io.micronaut.context.MessageSource;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.LocaleResolver;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import jakarta.inject.Singleton;
+
+import java.text.MessageFormat;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.micronaut.http.HttpStatus.*;
+
+/**
+ * It generates HTML error response page for a given {@link HttpStatus}.
+ * @author Sergio del Amo
+ * @since 4.7.0
+ */
+@Internal
+@Singleton
+final class DefaultHtmlErrorResponseBodyProvider implements HtmlErrorResponseBodyProvider<String> {
```

Comment:
Maybe all the templates can be a property, so a user can override it?

Reply:
I made the class internal. If I want to the class extendable I would make it non internal. For now, I think you can provide your own implementation of `HtmlErrorResponseBodyProvider` if you want to customize it. Or provide localization for other languages.